logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/RegisterMLSClientUseCase.kt

```diff
@@ -53,23 +54,33 @@ internal class RegisterMLSClientUseCaseImpl(
     private val userConfigRepository: UserConfigRepository
 ) : RegisterMLSClientUseCase {
 
-    override suspend operator fun invoke(clientId: ClientId): Either<CoreFailure, RegisterMLSClientResult> =
-        userConfigRepository.getE2EISettings().flatMap { e2eiSettings ->
+    override suspend operator fun invoke(clientId: ClientId): Either<CoreFailure, RegisterMLSClientResult> {
+        kaliumLogger.d("Registering MLS client with client ID: $clientId")
+        return userConfigRepository.getE2EISettings().flatMap { e2eiSettings ->
+            kaliumLogger.d("Registering MLS client with E2EI settings")
             if (e2eiSettings.isRequired && !mlsClientProvider.isMLSClientInitialised()) {
+                kaliumLogger.d("E2EI settings are required but MLS client is not initialised")
                 return RegisterMLSClientResult.E2EICertificateRequired.right()
             } else {
                 mlsClientProvider.getMLSClient(clientId)
             }
         }.onFailure {
+            kaliumLogger.e("Failed to get MLS client: $it")
```

Comment:
this is a wrong error, this case happens when e2ei is disabled or can't be fetched, then we proceed with normal mls client registeration flow.